common-content/en/module/js1/anonymous-functions/index.md

```diff
@@ -39,7 +39,7 @@ Stop and identify the difference in syntax between these two function definition
 
 We didn't give a name to the function in our Jest test.
 
-This is ok, because we don't need it to have a name. We don't call the function by name. We passed the function as an argument to the `test` function. The `test` function takes the function as a parameter. And function parameters get their own names in the scope of the function.
+This is ok, because we don't need it to have a name. We don't call the function by name. We passed the function as an {{<tooltip title="argument">}}Arguments are values given to a function which can be different every time we call the function.{{</tooltip>}} to the `test` function. The `test` function takes the function as a {{<tooltip title="parameter">}}A parameter is a named variable inside a function. The variable's value is given by the caller, when the function is called.{{</tooltip>}}. And function parameters get their own names in the {{<tooltip title="scope">}}Scope is where a variable can be accessed from. When we define function, its parameters are only available inside the function.{{</tooltip>}} of the function.
```

Comment:
This is a thought that I don't know the answer to so I invite you to think it too.

We have defined all these terms before, eg 

https://programming.codeyourfuture.io/structuring-data/sprints/1/prep/#parameters 
A parameter is a special kind of variable: its value is defined by the caller.
🧶 argument
Arguments are inputs given to a function inside (). An argument means an input.

https://programming.codeyourfuture.io/structuring-data/sprints/1/prep/#scope

etc

Do we think it's useful to always use the same def, or keep complicating it, or some mixture of the two?

Reply:
Yeah, I've have this thought too and am pretty conflict about it.

I think in general being able to hark back to a "general purpose definition" is useful in 80% of the cases (and when this happens, I exactly copy+paste tooltips and am sad it's unmaintained), but I consciously rewrote these rather than copying them because I thought it was more clear in context (and that there's hopefully more understanding now than when we first introduced them)...

Basically I think there are three contexts for a definitional tool-tip:
1. General purpose
2. Simplified because it's a first introduction and needs simplification.
3. Special-cased because something specific is being pointed out / referenced that wouldn't _generally_ be useful in a definition.

Which leans me to: It would probably be useful to have a central glossary-ish thing somewhere that can be invoked through a shortcode, and I'd probably use it 80% of the time, but it doesn't feel as high priority a thing to build as it would if I would use it 99% of the time...

Reply:
Forked off discussion into https://github.com/CodeYourFuture/curriculum/issues/1083